src/nxt_h1proto.c

```diff
@@ -884,6 +882,34 @@ nxt_h1p_request_body_read(nxt_task_t *task, nxt_http_request_t *r)
         break;
     }
 
+    if (r->chunked) {
+        h1p->chunked_parse.mem_pool = r->mem_pool;
+        in = h1p->conn->read;
+        size = nxt_buf_mem_used_size(&in->mem);
+        if (size == 0) {
+            status = NXT_HTTP_NOT_IMPLEMENTED;
+            goto error;
+        }
+
+        nxt_buf_t *out = nxt_http_chunk_parse(task, &h1p->chunked_parse, in);
+
+        if (h1p->chunked_parse.chunk_error ||
+ h1p->chunked_parse.error) {
+            status = NXT_HTTP_NOT_IMPLEMENTED
+;
+            goto error;
+        }
+
+        if (h1p->chunked_parse.last) {
+            out->next = nxt_buf_sync_alloc(r->mem_pool, NXT_BUF_SYNC_LAST);
+            out = nxt_h1p_chunk_create(task, r, out);
+            r->body = out;
+            r->state->ready_handler(task, r,
+NULL);
```

Comment:
... same as here ...

But this is clearly just a PoC/work in progress...

Reply:
Yep, just for PoC.